### DISCUSSION TITLE
specific image in batch

### DISCUSSION THREAD

**Discussion by SimonStochholm** [ultralytics/ultralytics#9195] [Docs]
Hi, I'm trying to modify yolov8 locally, and so during the training process I need to know on which specific image detections were done. Is there a way to know that specific image?
The variable: batch['im_file'] holds all image paths in the current batch, but I need to index into the one whose predictions are being tested.

**Comment by pderrenger** [comment]
Hi there! 👋

Absolutely, you can index into the specific images within a batch using the index of the detection or prediction you're investigating. If you have the index of the detection, let's call it `detection_index`, you can find the corresponding image path like so:

```python
image_path = batch['im_file'][detection_index]
```

`detection_index` should match the index of the predictions array you're looking at. This way, you can directly correlate your detections or predictions back to the specific images they were found in.

Hope this helps, and happy coding with YOLOv8! 🚀

**Reply by SimonStochholm** [reply]
Thank you. It turns out the specific index I was looking for is called "si" in the code.

**Reply by pderrenger** [reply]
I'm glad you found what you were looking for! If you have any more questions or need further assistance, don't hesitate to reach out. Happy coding with YOLOv8! 🚀
